src/sqlfluff/core/parser/markers.py

```diff
@@ -1,190 +1,205 @@
-                max(
-                    other.source_slice.stop
-                    for other in others
-                    if hasattr(other, "templated_slice")
-                ),
-            ),
-            is_literal=all(getattr(other, "is_literal", True) for other in others),
-            source_pos_marker=self.source_pos_marker,
+    def start_point_marker(self) -> "PositionMarker":
+        """Get a point marker from the start."""
+        return self.__class__.from_point(
+            self.source_slice.start,
+            self.templated_slice.start,
+            templated_file=self.templated_file,
+            # Start points also pass on the working position.
+            working_line_no=self.working_line_no,
+            working_line_pos=self.working_line_pos,
         )
 
-    def strip(self):
-        """Strip back anything enriched from this position marker."""
-        return FilePositionMarker(
-            statement_index=self.statement_index,
-            line_no=self.line_no,
-            line_pos=self.line_pos,
-            char_pos=self.char_pos,
+    def end_point_marker(self) -> "PositionMarker":
+        """Get a point marker from the end."""
+        return self.__class__.from_point(
+            self.source_slice.stop,
+            self.templated_slice.stop,
+            templated_file=self.templated_file,
         )
+
+    @staticmethod
+    def slice_is_point(test_slice):
+        """Is this slice a point."""
+        return test_slice.start == test_slice.stop
+
+    def is_point(self) -> bool:
+        """A marker is a point if it has zero length in templated and source file."""
+        return self.slice_is_point(self.source_slice) and self.slice_is_point(
+            self.templated_slice
+        )
+
+    @staticmethod
+    def infer_next_position(raw: str, line_no: int, line_pos: int) -> Tuple[int, int]:
+        """Using the raw string provided to infer the position of the next."""
+        if not raw:
+            return line_no, line_pos
+        split = raw.split("\n")
+        return (
+            line_no + len(split) - 1,
+            line_pos + len(raw) if len(split) == 1 else len(split[-1]),
+        )
+
+    def with_working_position(self, line_no: int, line_pos: int):
+        """Copy this position and replace the working position."""
+        return self.__class__(
+            source_slice=self.source_slice,
+            templated_slice=self.templated_slice,
+            templated_file=self.templated_file,
+            working_line_no=line_no,
+            working_line_pos=line_pos,
+        )
+
+    def is_literal(self) -> bool:
+        """Infer literalness from context."""
+        # NOTE: We should define what the implication of is_literal is here.
+        # TODO: I think we were inconsistent in whether a zero length source slice
+        # is (or is not) literal. The lexer initially defined it as False, this
+        # function will say True. Not sure what the implications of that are yet.
```

Comment:
This is one of those moments when it feels somewhat comforting to have solid test coverage. 💪🏽

Reply:
I think I've flushed out all the the cases where this matters. I'll update this comment.